macros/is_table_empty.sql

```diff
@@ -0,0 +1,20 @@
+{%- macro is_table_empty(table_name) -%}
+
+{{ adapter.dispatch('is_table_empty', 'linkedin_source') (table_name) }}
+
+{%- endmacro %}
+
+{%- macro default__is_table_empty(table_name) -%}
+    {%- if execute and flags.WHICH in ('run', 'build') %}
+        {% set row_count_query %}
+            select count(*) as row_count from {{ table_name }}
```

Comment:
I'm wondering do we need to count the entire table if we're just checking if the table is empty? Not that there are many cases where there are a billion records in Linkedin that could impact query execution but wondering if this condition could be more performant.

Reply:
Interesting point 🤔. What would you recommend for adjusting this to count the entire table?

Reply:
@fivetran-joemarkiewicz Would something like

{%- macro default__is_table_empty(table_name) -%}
    {%- if execute and flags.WHICH in ('run', 'build') %}
        {% set row_check_query %}
            SELECT 1
            FROM {{ table_name }}
            LIMIT 1
        {% endset %}
        {% set results = run_query(row_check_query) %}
        {% if results and results.rows | length == 0 %}
            {{ return("empty") }} 
        {% endif %}
    {% endif -%}
{%- endmacro -%}


work? The length would be 1 if there's a row (and thus not empty), and 0 if not.

Reply:
I like this idea, but if we were to implement it then I'd want to validate this a bit more since it is more than just a one line code change.

Reply:
Sensible! Just a suggestion anyway, this PR is approved.

Reply:
@fivetran-avinash I thought more about this and wanted to make try some tests to see if we could get this more performant option working before release. I was able to get this working in the latest commit. See the validations below. Let me know your thoughts!

- ✅  When running on an empty schema, the macro works as expected.
![image](https://github.com/user-attachments/assets/fddedfcc-a6e7-49e6-8a81-e5a2c5710f59)

- ✅  When running on a populated schema, the macro works as expected.
![image](https://github.com/user-attachments/assets/ad9a573a-296d-442e-ab53-f16046193d55)

- ✅  When running on two populated union schemas, the macro works as expected.
![image](https://github.com/user-attachments/assets/39a610bc-094b-4fa8-80ae-b4e7cf6e1fee)

- ✅  When running on two union schemas which are empty, the macro works as expected.
![image](https://github.com/user-attachments/assets/7ab66eae-8d38-4146-bb94-675e176bd44f)

Would you be able to take a second look and let me know if you have any notes before this moves forward.

Reply:
I also reran the downstream integration tests and saw success. So it looks like we are in the clear that this doesn't cause downstream issues.

Reply:
Was able to replicate several of these setups, so I'd say it's safe to move forward.